src/main/java/org/spongepowered/api/item/inventory/Transformation.java

```diff
@@ -0,0 +1,91 @@
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.item.inventory;
+
+import org.spongepowered.api.CatalogType;
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.item.inventory.query.QueryOperation;
+import org.spongepowered.api.util.ResettableBuilder;
+import org.spongepowered.api.util.annotation.CatalogedBy;
+
+@CatalogedBy(Transformations.class)
+public interface Transformation extends CatalogType {
+
+    /**
+     * Transforms an Inventory
+     *
+     * @param inventory The inventory to transform
+     *
+     * @return The transformed inventory
+     */
+    Inventory transform(Inventory inventory);
+
+    static Builder builder() {
+        return Sponge.getRegistry().createBuilder(Builder.class);
+    }
+
+    /**
+     * Creates a new Transformation based on QueryOperations
+     *
+     * @param operations The QueryOperations
+     *
+     * @return The new transformation
+     */
+    static Transformation of(QueryOperation... operations) {
+        if (operations.length == 0) {
+            return Transformations.NONE;
+        }
+        return builder().append(operations).build();
+    }
+
+    interface Builder extends ResettableBuilder<Transformation, Builder> {
+
+        /**
+         * Appends one or more {@link QueryOperation}s
+         *
+         * @param operation The operations to add
+         *
+         * @return Fluent pattern
+         */
+        Builder append(QueryOperation... operation);
+
+        /**
+         * Registers the Transformation with given id
+         *
+         * @param id The registration id
+         *
+         * @return The registered Transformation
+         */
+        Transformation build(String id);
+
+        /**
+         * Builds the Transformation with a random id
```

Comment:
Random ids are worthless at best and harmful at worst.
All `CatalogType`s should have meaningful ids.